WordPress/Classes/WordPress.xcdatamodeld/WordPress 87.xcdatamodel/contents

```diff
@@ -344,6 +344,12 @@
         <attribute name="isPrimary" optional="YES" attributeType="Boolean" defaultValueString="NO" usesScalarValueType="NO" syncable="YES"/>
         <relationship name="blog" optional="YES" maxCount="1" deletionRule="Nullify" destinationEntity="Blog" inverseName="domains" inverseEntity="Blog" syncable="YES"/>
     </entity>
+    <entity name="FollowersStatsRecordValue" representedClassName=".FollowersStatsRecordValue" parentEntity="StatsRecordValue" syncable="YES">
```

Comment:
In `FollowersStatsRecordValue+CoreDataProperties.swift`, `name` is of type `String?`.
`name` is not designated optional here in the same fashion as `avatarURLString `.

Do you think it would be worthwhile to reconcile that difference?

Reply:
So, that‘s on purpose. Well, kinda. That‘s sorta limitation/design of Core Data and Swift interop. Changing the flag in .xcdatamodel enforces that the value is present at the point of _saving the changes to a `NSManagedObjectContext`_. Swift requires the value to be present _at all times_, which doesn‘t mesh well with some Core Data features like faulting and fetching only selected properties (https://developer.apple.com/documentation/coredata/nsfetchrequest/1506851-propertiestofetch). For that reason, Xcode / Core Data Model Editor won‘t ever generate a Swift class with non-optional, non-primitive properties. 

I want Core Data stack to yell at us here if we try to create a Followers record value with no `name` set.

Arguably, I could manually change the property type in generated Swift class manually — but it‘d get lost when someone generates it next, and because of the way CD works would be semantically incorrect (I could conceivably write a `NSFetchRequest	` that only fetches the `followersCount` property, though I have no idea why I would ever actually do that).

Reply:
Thanks for the detailed explanation. What is the drawback for `name` to just be type `String` (i.e., empty instead of `nil`)?

Reply:
@jklausa & I conferred offline, and he cleared up a misunderstanding on my part.